autoload/lsp.vim

```diff
@@ -250,10 +250,15 @@ function! s:on_text_document_did_open(...) abort
     call lsp#internal#diagnostics#state#_force_notify_buffer(l:buf)
 
     for l:server_name in lsp#get_allowed_servers(l:buf)
-        call s:ensure_flush(l:buf, l:server_name, function('s:fire_lsp_buffer_enabled', [l:server_name, l:buf]))
+        call s:ensure_flush(l:buf, l:server_name, function('s:done_text_document_did_open', [l:server_name, l:buf]))
     endfor
 endfunction
 
+function! s:done_text_document_did_open(server_name, buf, ...) abort
+    call lsp#ui#vim#semantic#semantic_full(a:server_name, a:buf)
```

Comment:
instead of firing it here. would it be possible to do something similar to [document highlighting]( https://github.com/prabirshrestha/vim-lsp/blob/master/autoload/lsp/internal/document_highlight.vim).

This allows granular control such as `g:lsp_semantic_delay_highlight`

Reply:
`lsp#send_request` will automatically flush if it is out of date and also helps keeping the flush fast.

Reply:
Done in [1d11c6d](https://github.com/prabirshrestha/vim-lsp/pull/1275/commits/1d11c6dacb00d96585693890677be0499bec97c3)